### DISCUSSION TITLE
Select table of indexes

### DISCUSSION THREAD

**Discussion by xav-ie** [questdb/questdb#2030] [Ideas]
According to https://questdb.io/docs/concept/indexes/, whenever you make a column indexed, it creates a second table of the symbols you are indexing, one row for each unique symbol. What I am hoping to do is just to simply select all of the rows in this second table and get the symbols. Is there a way to do this?

What I am doing right now is `SELECT DISTINCT sym FROM 'mytable';`. I am not sure if this is the best way to do this, but it does execute very fast... is this the proper way? Would it be a good idea to make it possible for users to select from this index table?

**Comment by puzpuzpuz** [comment]
QuestDB's indexes are basically inverted indexes, i.e. for a given symbol value they provide a list of row ids for the rows that have the symbol value. They're not a separate table as such, but an internal persistent data structure.

Also, non-indexed symbols themselves assume a key-value (or, a dictionary) mapping between the string symbol representation and the internal id (which is an `int`). So, technically speaking an index is not necessary to get a list of possible symbol values thanks to symbol being a dictionary type.

> I am not sure if this is the best way to do this, but it does execute very fast... is this the proper way?

Yes, your query is the right way to fetch the list of symbol values.
